lib/chef/resource/systemd_service.rb

```diff
@@ -0,0 +1,49 @@
+#
+# Author:: Davide Cavalca (<dcavalca@fb.com>)
+# Copyright:: Copyright (c) 2015 Facebook
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/resource/service'
+require 'chef/provider/service/systemd'
+
+class Chef
+  class Resource
+    class SystemdService < Chef::Resource::Service
+      provides :service, os: "linux" do |node|
+        Chef::Platform::ServiceHelpers.service_resource_providers.include?(:systemd)
+      end
```

Comment:
OK, so what's the right way to do that?

`yum_package` **resource** also responds to `package`:

```
    class YumPackage < Chef::Resource::Package

      provides :yum_package
      provides :package, os: "linux", platform_family: [ "rhel", "fedora" ]
```

so that you can allow_downgrade. You're saying you don't want stuff but you're not telling us what you DO want.

`service` as a resource needs to work on systemd systems and have the right actions available without people having to specify `systemd_service`. It's unclear what mechanism you want Davide to use to make that happen.